### PR TITLE
[DPE-5032] Add 3.5 release bundle

### PR DESCRIPTION
**Pull request by deusebio**

### CODE REVIEW

**Review by theoctober19th — APPROVED** [review]

File: .github/workflows/ci-tests.yaml [hunk 1]
```diff
@@ -50,6 +50,8 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
+        version:
+          - --spark-version "3.5.1"
```
Comment (welpaolo) [inline .github/workflows/ci-tests.yaml:54]: Do we want to run tests only on 3.5.X or also on 3.4.X ?

File: python/spark_test/fixtures/pod.py [hunk 2]
```diff
@@ -20,20 +20,34 @@ def admin_pod_name():
 
 
 @pytest.fixture(scope="module")
-def admin_pod(kubeconfig, namespace, admin_pod_name):
+def spark_version():
+    return "3.4.2"
```
Comment (welpaolo) [inline python/spark_test/fixtures/pod.py:24]: Here should be 3.5.1 ?

File: releases/3.5/terraform/cos/applications.tf [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+  resources = {
+    git-sync-image = 32
+  }
+
+  config = {
+    git_branch = "main"
+    git_depth  = "1"
+    git_repo   = "https://github.com/canonical/spark-k8s-bundle"
+    grafana_dashboards_path = "releases/3.4/resources/grafana/"
```
Comment (welpaolo) [inline releases/3.5/terraform/cos/applications.tf:44]:
```suggestion
    grafana_dashboards_path = "releases/3.5/resources/grafana/"
```

**Review by welpaolo — APPROVED** [review]
LGTM, I left some comments

File: .github/workflows/ci-tests.yaml (hunk 1 above)
Comment (deusebio) [reply .github/workflows/ci-tests.yaml:54]: I believe so. But I wanted first to make sure that the tests were working on 3.4.2. Let me update this, but the flakiness of the tests is a bit problematic here.

File: python/spark_test/fixtures/pod.py (hunk 2 above)
Comment (deusebio) [reply python/spark_test/fixtures/pod.py:24]: This should not matter too much, as this is generally overridden by the flag provided in the input.

File: releases/3.5/terraform/cos/applications.tf (hunk 3 above)
Comment (deusebio) [reply releases/3.5/terraform/cos/applications.tf:44]: Good catch!

**Comment by deusebio** [comment]
As discussed, I'm going to merge, although tests are VERY flaky (failed tests needed to be re-run 6 times, to get them all to pass). We will address the flakyness in the [DPE-5078](https://warthogs.atlassian.net/browse/DPE-5078).

**Pull request merged**